cmd/fleetctl/fleetctl/gitops_test.go

```diff
@@ -3084,3 +3086,195 @@ func TestGitOpsNoTeamConditionalAccess(t *testing.T) {
+						Token: "test-token",
+					}, nil
+				}
+			},
+			dryRunAssertion: func(t *testing.T, out string, err error) {
+				assert.NoError(t, err)
+				assert.Contains(t, out, "[+] would've applied EULA")
+			},
+			realRunAssertion: func(t *testing.T, out string, err error) {
+				assert.NoError(t, err)
+				assert.Contains(t, out, "[+] applied EULA")
+			},
+		},
+		{
+			name: "no EULA specified (no existing EULA uploaded)",
+			cfg:  createGlobalGitOpsConfig(""),
+			mockSetup: func(t *testing.T, ds *mock.Store) {
+				ds.MDMGetEULAMetadataFunc = func(ctx context.Context) (*fleet.MDMEULA, error) {
+					return nil, &notFoundError{} // No existing EULA
+				}
+			},
+			dryRunAssertion: func(t *testing.T, out string, err error) {
+				assert.NoError(t, err)
+				assert.Contains(t, out, "[+] would've applied EULA")
+			},
+			realRunAssertion: func(t *testing.T, out string, err error) {
+				assert.NoError(t, err)
+				assert.Contains(t, out, "[+] applied EULA")
+			},
+		},
+		{
+			name: "deleting existing EULA",
+			cfg:  createGlobalGitOpsConfig(""),
+			mockSetup: func(t *testing.T, ds *mock.Store) {
+				ds.MDMGetEULAMetadataFunc = func(ctx context.Context) (*fleet.MDMEULA, error) {
+					return &fleet.MDMEULA{
+						Name:  pdfPath,
+						Token: "test-token",
+					}, nil
+				}
+			},
+			dryRunAssertion: func(t *testing.T, out string, err error) {
+				assert.NoError(t, err)
+				assert.Contains(t, out, "[+] would've applied EULA")
+			},
+			realRunAssertion: func(t *testing.T, out string, err error) {
+				assert.NoError(t, err)
+				assert.Contains(t, out, "[+] applied EULA")
+			},
+		},
+		{
+			name: "not a PDF file",
+			cfg:  createGlobalGitOpsConfig(fmt.Sprintf(`end_user_license_agreement: "%s"`, invalidPDFPath)),
+			mockSetup: func(t *testing.T, ds *mock.Store) {
+				ds.MDMGetEULAMetadataFunc = func(ctx context.Context) (*fleet.MDMEULA, error) {
+					return nil, &notFoundError{} // No existing EULA
+				}
+			},
+			dryRunAssertion: func(t *testing.T, out string, err error) {
+				assert.ErrorContains(t, err, "invalid file type")
+			},
+			realRunAssertion: func(t *testing.T, out string, err error) {
+				assert.ErrorContains(t, err, "invalid file type")
+			},
+		},
+		{
+			name: "uploading the same EULA again",
+			cfg:  createGlobalGitOpsConfig(""),
+			mockSetup: func(t *testing.T, ds *mock.Store) {
+				ds.MDMGetEULAMetadataFunc = func(ctx context.Context) (*fleet.MDMEULA, error) {
+					hash := sha256.Sum256(pdfContent) // Simulate same EULA
+					return &fleet.MDMEULA{
+						Name:   pdfPath,
+						Token:  "test-token",
+						Sha256: hash[:], // Simulate same EULA
+					}, nil
+				}
+			},
+			dryRunAssertion: func(t *testing.T, out string, err error) {
+				assert.NoError(t, err)
+				assert.Contains(t, out, "[+] would've applied EULA")
+			},
+			realRunAssertion: func(t *testing.T, out string, err error) {
+				assert.NoError(t, err)
+				assert.Contains(t, out, "[+] applied EULA")
+			},
+		},
+	}
```

Comment:
Nice! Just wondering if it would be possible to assert something more in the real assertion, e.g. whether or not the save EULA mock datastore method was called? (would need to be reset after each test case)

Reply:
yep, added some new assertions